sympy/tensor/array/ndim_array.py

```diff
@@ -367,19 +367,29 @@ def __sub__(self, other):
 
     def __mul__(self, other):
         from sympy.matrices.matrices import MatrixBase
+        from sympy.tensor.array import SparseNDimArray
 
         if isinstance(other, (Iterable, NDimArray, MatrixBase)):
             raise ValueError("scalar expected, use tensorproduct(...) for tensorial product")
+
         other = sympify(other)
+        if isinstance(self, SparseNDimArray):
+            return type(self)({k: other*v for (k, v) in self._sparse_array.items()}, self.shape)
```

Comment:
When it is known that `type(self)` is `SparseNDimArray` then why not to directly use something like,
`SparseNDimArray({k: other*v for (k, v) in self._sparse_array.items()}, self.shape)`?

Reply:
Because the constructor of `SparseNDimArray`  will initialize a `ImmutableSparseNDimArray`. But if it is a MutableArray who triggers this function, il would be better to keep the output as mutable.

Reply:
In other word, `SparseNDimArray` is like a generic class where we define some common behavior. But then there are `ImmutableSparseNDimArray` and `MutableSparseNDimArray` as subclass.